src/amber/dsl/router.cr

```diff
@@ -16,7 +16,8 @@ module Amber::DSL
         content = controller.{{ action.id }}
         controller.run_after_filter(action)
         controller.run_after_filter(:all)
-        content
+        context.response.print(content)
+        context.response.close
```

Comment:
You could probably signal "halting" in a different way, maybe a custom property on Response or Context...